File: Arrays/LargestSumContinous/LargestSumContinous.java
package Arrays.LargestSumContinous;

public class LargestSumContinous {

    public static int largestSumContinous(int[] arr){
        int sum=Integer.MIN_VALUE,temp=0;
        for(int no:arr){
            if(temp+no<0){
                temp=0;
                sum = Math.max(sum, no);
            }else{
                temp += no;
                sum = Math.max(sum, temp);
            }
        }
        return sum;
    }
    
    public static void main(String[] args){
        int[] arr = {-1,-2,-3,-4};
        System.out.println(largestSumContinous(arr));
    }
}

File: Arrays/LargestSumContinous/LargestSumContinous.go
package main

import "fmt"

func largestSumContious(arr []int) int {
	sum := arr[0]
	temp := arr[0]
	n := len(arr)
	for i := 1; i < n; i++ {
		if temp+arr[i] < 0 {
			temp = 0
			if arr[i] > sum {
				sum = arr[i]
			}
		} else {
			temp += arr[i]
			if temp > sum {
				sum = temp
			}
		}
	}
	return sum
}

func main() {
	arr := []int{-1, -2, -3, -4}
	fmt.Println(largestSumContious(arr))
}
